### PR TITLE
[tesla] Introduce "allowWakeUpForCommands" option

### PR DESCRIPTION
**Pull request by kaikreuzer**
This introduces a new configuration option to the vehicle thing, which allows the binding to wake up the vehicle in case that commands are sent to it.
This is specifically handy for things like activating the auto-conditioning, Sentry mode or similar things.
As waking up the vehicle usually takes ~15 seconds, sending the command is retried 3 times (with 15 secs pause) until it succeeds. Naturally, this means that the command is not executed immediately, but that it is delayed by a considerable time.

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaAccountHandler.java [hunk 1]
```diff
@@ -308,7 +308,8 @@ private ThingStatusInfo authenticate() {
         return new ThingStatusInfo(ThingStatus.ONLINE, ThingStatusDetail.NONE, null);
     }
 
-    protected String invokeAndParse(String vehicleId, String command, String payLoad, WebTarget target) {
+    protected String invokeAndParse(String vehicleId, String command, String payLoad, WebTarget target,
+            int noOfretries) {
```
Comment (lolodomo) [inline bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaAccountHandler.java:312]: Typo? nbOfretries?

File: bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaAccountHandler.java [hunk 2]
```diff
@@ -445,16 +456,21 @@ private String getBasicAuthentication() {
 
     protected class Request implements Runnable {
 
+        private static final int NO_OF_RETRIES = 3;
```
Comment (lolodomo) [inline bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaAccountHandler.java:459]: NB?

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaAccountHandler.java (hunk 1 above)
Comment (jlaur) [reply bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaAccountHandler.java:312]: _No._ is the correct abbreviation for _Number_. So noOf**R**etries or **number**Of**R**etries for perfection. :)

Comment (lolodomo) [reply bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaAccountHandler.java:312]: In French, the abbreviation for number is nb, I thought it was the same in English.

Comment (kaikreuzer) [reply bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaAccountHandler.java:312]: See https://en.wikipedia.org/wiki/Numero_sign

File: bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaAccountHandler.java (hunk 2 above)
Comment (kaikreuzer) [reply bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaAccountHandler.java:459]: See above.

File: bundles/org.openhab.binding.tesla/src/main/resources/OH-INF/thing/modely.xml [hunk 3]
```diff
@@ -136,6 +136,12 @@
 				<label>Allow Wake-Up</label>
 				<description>Allows waking up the vehicle. Caution: This can result in huge vampire drain!</description>
 			</parameter>
+			<parameter name="allowWakeupForCommands" type="boolean" required="false">
+				<default>false</default>
+				<label>Allow Wake-Up For Commands</label>
```
[the remark this answers is not in the document]
Comment (kaikreuzer) [reply bundles/org.openhab.binding.tesla/src/main/resources/OH-INF/thing/modely.xml:141]: Done, thanks for noticing, I always forget...

**Review by lolodomo — APPROVED** [review]
LGTM

**Pull request merged**